mesonbuild/backend/ninjabackend.py

```diff
@@ -2213,6 +2213,10 @@ def generate_swift_target(self, target):
         compile_args = self.generate_basic_compiler_args(target, swiftc)
         compile_args += swiftc.get_compile_only_args()
         compile_args += swiftc.get_module_args(module_name)
+        if mesonlib.version_compare(swiftc.version, '>=5.9'):
+            compile_args += swiftc.get_cxx_interoperability_args(target.compilers)
```

Comment:
This will add those flags on _every_ compilation. Is there ever a case where users would not want that to happen?

Reply:
It looks like it adds it whenever a specific target has source files for both languages. So my assumption is that any situation where this PR adds the flags, was previously a link error.

@tangalbert919 please correct me if I misunderstood.

Reply:
Another question that came to mind is what happens if you have lib A that is in C++, lib B that is in Swift and exe C that links with both A and B (and let's say B uses C++ methods from A). Do you need to have that flag when compilng B? What about when compiling C? Does it matter whether C uses methods from A or not?

Reply:
When compiling lib B, you must specify the flag so it can even find the C++ methods it intends to use from lib A. Exe C also needs that flag if it is using C++ methods from lib A as well, assuming C is a Swift executable.

Reply:
I'd like to see a test for this then, please

Reply:
I can modify one of the test cases I made to handle the scenario you had thought of, or I can create a new test case entirely.